cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py

```diff
@@ -183,14 +182,64 @@ def _xx_yy_zz_interaction_via_full_czs(q0: cirq.Qid, q1: cirq.Qid, x: float, y:
 
 
 def cleanup_operations(operations: Sequence[ops.Operation]):
+    operations = _merge_single_qubit_gates(operations)
     circuit = circuits.Circuit(operations)
-    circuit = merge_single_qubit_gates_to_phased_x_and_z(circuit)
     circuit = eject_phased_paulis(circuit)
     circuit = eject_z(circuit)
     circuit = circuits.Circuit(circuit.all_operations(), strategy=circuits.InsertStrategy.EARLIEST)
     return list(circuit.all_operations())
 
 
+def _transform_single_qubit_operations_to_phased_x_and_z(
+    operations: Sequence[ops.Operation],
+) -> Sequence[ops.Operation]:
+    """Transforms operations on the same qubit to a PhasedXPowGate followed by a Z gate.
+
+    Args:
+        operations: sequence of operations on the same qubit
+    Returns:
+        A PhasedXPowGate followed by a Z gate. If one the gates is not needed, it will be omitted.
+    """
+    u = np.eye(2)
+    for op in operations:
+        u = protocols.unitary(op) @ u
+    return [
+        g(op.qubits[0])
+        for g in single_qubit_decompositions.single_qubit_matrix_to_phased_x_z(u, atol=1e-8)
+    ]
+
+
+def _merge_single_qubit_gates(operations: Sequence[ops.Operation]) -> Sequence[ops.Operation]:
```

Comment:
can you make this a public method in a seprate PR, seems like a useful transformation?